documentation/design/user-stories/shared-infra.adoc

```diff
@@ -0,0 +1,307 @@
+metadata:
+  name: mydlq
+  namespace: myapp
+spec:
+  systemQueue: {}
+---
+apiVersion: enmasse.io/v1beta2
+kind: MessagingAddress
+metadata:
+  name: myexpiry
+   namespace: myapp
+spec:
+  systemQueue: {}
+```
+
+Then create
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingAddress
+metadata:
+  name: myqueue
+  namespace: myapp
+spec:
+  queue:
+    # Note: a queue 'mydlq' must exist!
+    DeadLetterQueue: mydlq
+    # Note: a queue 'myexpiry' must exist!
+    ExpiryQueue: myexpiry
+
+    # Forwarders require a MessagingConnector to exist, in this case one that is named 'remote1'.
+    forwarders:
+    - name: fwd1
+      remoteAddress: remote1/queue1
+      direction: in
+```
+
+NOTE: the properties that are only applicable to a few types, are specified under the type object as shown above.
+
+=== CRUDLW a messaging endpoint in a namespace where messaging is enabled
+
+Create a `MessagingEndpoint` resource containing either `cluster`, `route`, `ingress`, `loadbalancer` or `nodeport` objects
+that specify the properties. The protocols supported are AMQP, AMQP+TLS,
+AMQP-WS and AMQP-WS+TLS.
+
+The default protocols depends on the endpoint type. For "Ingress" and "Route", the default
+protocols is AMQPS. For NodePort, LoadBalancer and Cluster, the default protocols are AMQP, AMQPS.
+
+==== Configure an in-cluster endpoint with TLS
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingEndpoint
+metadata:
+  name: endpoint1
+  namespace: myapp
+spec:
+  # TLS configuration for endpoint
+  tls: {}
+
+  protocols:
+    - AMQPS
+    - AMQP
+
+  cluster: {}
+```
+
+==== Configure an endpoint exposed as a route
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingEndpoint
+metadata:
+  name: endpoint1
+  namespace: myapp
+spec:
+  route:
+    hostname: messaging.example.com # Optional
+```
+
+==== Configure an endpoint exposed as a loadbalancer
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingEndpoint
+metadata:
+  name: endpoint1
+  namespace: myapp
+spec:
+  tls: {}
+  loadbalancer
+    annotations: {}
+    requireTls: true # Setting this to 'true' will disable non-TLS port
+```
+
+==== Status section
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingEndpoint
+metadata:
+  name: endpoint1
+  namespace: myapp
+spec:
+  ...
+status:
+  phase: Active
+  host: endpoint1-myapp.enmasse-infra.svc
+  ports:
+  - name: "AMQP"
+    value: 5672
+  - name: "AMQP+TLS"
+    value: 5671
+  tls:
+    selfsigned:
+      ca: # CA PEM
+  conditions: # Conditions array for kubernetes-style conditions
+```
+
+=== CRUDLW a messaging connector in a namespace where messaging is enabled
+
+=== Managing certificates for a messaging endpoint
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingEndpoint
+metadata:
+  name: endpoint1
+  namespace: myapp
+spec:
+  ...
+
+  # Required for endpoints of type 'route' and 'ingress', as these only support TLS
+  # For other types, specifying `tls` will ensure they expose traffic on TLS
+  tls:
```

Comment:
A tis endpoint should be configurable to want or need a client certificate.  We need configuration for this use case together with configuration for the trusted CA anchor(s)/intermediates.